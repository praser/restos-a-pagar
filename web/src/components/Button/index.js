import styled from 'styled-components';

export const Button = styled.button`
  font-size: 0.8rem;
  border-radius: 0.35rem;
  padding: 0.75rem 1rem;
  color: rgba(255, 255, 255, 0.8);
  font-weight: 400;
  text-align: center;
  border: 1px solid transparent;
  line-height: 1.5;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out,
    -webkit-box-shadow 0.15s ease-in-out;
  cursor: pointer;
  box-shadow: 0 0.125rem 0.25rem 0 rgba(58, 59, 69, 0.2);
  margin: 0 0.15rem;
  letter-spacing: 0.04rem;
  text-decoration: none;
  svg {
    margin-right: 0.3rem;
  }

  &:hover {
    background-color: #2e59d9;
    border-color: #2653d4;
    text-decoration: none;
    color: rgba(255, 255, 255, 1);
  }

  &:focus {
    outline: 0;
    box-shadow: 0 0 0 0.2rem rgba(105, 136, 228, 0.5);
  }

  &:disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }
`;

export const ButtonPrimary = styled(Button)`
  background-color: #4e73df;
  border-color: #4e73df;

  &:hover {
    background-color: #2e59d9;
    border-color: #2653d4;
  }
`;

export const ButtonDanger = styled(Button)`
  background-color: #ea4335;
  border-color: #ea4335;

  &:hover {
    background-color: #e12717;
    border-color: #e12717;
  }

  &&:focus {
    box-shadow: 0 0 0 0.2rem rgba(255, 39, 23, 0.5);
  }
`;

export const ButtonSecondary = styled(Button)`
  background-color: #36b9cc;
  border-color: #36b9cc;

  &:hover {
    background-color: #2c9faf;
    border-color: #2a96a5;
  }

  &&:focus {
    box-shadow: 0 0 0 0.2rem rgba(44, 159, 175, 0.5);
  }
`;

export const SmallButtonPrimary = styled(ButtonPrimary)`
  padding: 0.25rem 0.5rem;
  font-size: 0.875rem;
  line-height: 1.5;
  border-radius: 0.2rem;
`;

export const SmallButtonSecondary = styled(ButtonSecondary)`
  padding: 0.25rem 0.5rem;
  font-size: 0.875rem;
  line-height: 1.5;
  border-radius: 0.2rem;
`;
