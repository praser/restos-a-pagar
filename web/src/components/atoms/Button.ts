import styled from 'styled-components';

interface IProps {
  secondary?: boolean;
  warning?: boolean;
  danger?: boolean;
  small?: boolean;
}

const Button = styled.button<IProps>`
  background-color: #4e73df;
  border-color: #4e73df;
  font-size: 0.8rem;
  border-radius: 0.35rem;
  padding: 0.75rem 1rem;
  color: rgba(255, 255, 255, 0.8);
  font-weight: 400;
  text-align: center;
  border: 1px solid transparent;
  line-height: 1.5;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out,
    -webkit-box-shadow 0.15s ease-in-out;
  cursor: pointer;
  box-shadow: 0 0.125rem 0.25rem 0 rgba(58, 59, 69, 0.2);
  margin: 0 0.15rem;
  letter-spacing: 0.04rem;
  text-decoration: none;
  svg {
    margin-right: 0.3rem;
  }

  &:hover {
    background-color: #2e59d9;
    border-color: #2653d4;
    text-decoration: none;
    color: rgba(255, 255, 255, 1);
  }

  &:focus {
    outline: 0;
    box-shadow: 0 0 0 0.2rem rgba(105, 136, 228, 0.5);
  }

  &:disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }

  ${({ secondary }) =>
    secondary &&
    `
    background-color: #36b9cc;
    border-color: #36b9cc;

    &:hover {
      background-color: #2c9faf;
      border-color: #2a96a5;
    }

    &&:focus {
      box-shadow: 0 0 0 0.2rem rgba(44, 159, 175, 0.5);
    }
  `}

  ${({ warning }) =>
    warning &&
    `
    background-color: #f6c23e;
    border-color: #f6c23e;

    &:hover {
      background-color: #f4b619;
      border-color: #f4b30d;
    }

    &:focus {
      box-shadow: 0 0 0 0.2rem rgba(247, 203, 91, 0.5);
    }
  `}

  ${({ danger }) =>
    danger &&
    `
    background-color: #ea4335;
    border-color: #ea4335;

    &:hover {
      background-color: #e12717;
      border-color: #e12717;
    }

    &:focus {
      box-shadow: 0 0 0 0.2rem rgba(255, 39, 23, 0.5);
    }
  `}
  
  ${({ small }) =>
    small &&
    `
    padding: 0.25rem 0.5rem;
    font-size: 0.875rem;
    line-height: 1.5;
    border-radius: 0.2rem;
  `}
`;

const ButtonPrimary = styled(Button)``;
const ButtonDanger = styled(Button)``;
const ButtonSecondary = styled(Button)``;
const ButtonWarning = styled(Button)``;
const SmallButtonPrimary = styled(ButtonPrimary)``;
const SmallButtonSecondary = styled(ButtonSecondary)``;
const SmallButtonDanger = styled(ButtonDanger)``;
const SmallButtonWarning = styled(ButtonWarning)``;

export default Button;
