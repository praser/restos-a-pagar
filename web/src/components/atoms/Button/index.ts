import Button from './Buttom';

export default Button;
