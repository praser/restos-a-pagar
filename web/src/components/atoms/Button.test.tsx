import React from 'react';

import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import Button from 'components/atoms/Button';

describe('Button atom component', () => {
  it('is expected to render the component correctly', () => {
    const button = renderer.create(<Button>Teste</Button>).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('is expected to have a blue background by default', () => {
    const { container } = render(<Button>Teste</Button>);
    expect(container.firstChild).toHaveStyle('background-color: #4e73df');
  });

  it('is expected to have a gray background when secondary', () => {
    const { container } = render(<Button secondary>Teste</Button>);
    expect(container.firstChild).toHaveStyle('background-color: #36b9cc');
  });

  it('is expected to have a yellow background when warning', () => {
    const { container } = render(<Button warning>Teste</Button>);
    expect(container.firstChild).toHaveStyle('background-color: #f6c23e');
  });

  it('is expected to have a red background when danger', () => {
    const { container } = render(<Button danger>Teste</Button>);
    expect(container.firstChild).toHaveStyle('background-color: #ea4335');
  });

  it('is expected to be smaller when small', () => {
    const { container } = render(<Button small>Teste</Button>);
    expect(container.firstChild).toHaveStyle('padding: 0.25rem 0.5rem');
    expect(container.firstChild).toHaveStyle('font-size: 0.875rem');
    expect(container.firstChild).toHaveStyle('line-height: 1.5');
    expect(container.firstChild).toHaveStyle('border-radius: 0.2rem');
  });
});
