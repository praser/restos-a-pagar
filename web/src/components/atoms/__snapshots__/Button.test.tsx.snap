// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button atom component is expected to render the component correctly 1`] = `
<button
  className="sc-bdfBwQ hXEZTf"
/>
`;
